src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTestCase.php

```diff
@@ -1060,25 +1061,54 @@ public function testMessengerWithMultipleBuses()
 
         $this->assertTrue($container->has('messenger.bus.commands'));
         $this->assertSame([], $container->getDefinition('messenger.bus.commands')->getArgument(0));
-        $this->assertEquals([
-            ['id' => 'add_bus_name_stamp_middleware', 'arguments' => ['messenger.bus.commands']],
-            ['id' => 'reject_redelivered_message_middleware'],
-            ['id' => 'dispatch_after_current_bus'],
-            ['id' => 'failed_message_processing_middleware'],
-            ['id' => 'send_message', 'arguments' => [true]],
-            ['id' => 'handle_message', 'arguments' => [false]],
-        ], $container->getParameter('messenger.bus.commands.middleware'));
+
+        if (class_exists(LockMiddleware::class)) {
```

Comment:
Don't we also need to check that the lock component is installed?

Reply:
It's installed, we're in the test directory and `symfony/lock` is in require-dev.

Reply:
I mean on a user app. THe current will enable the middleware even if the lock component is not installed.

Reply:
We're on the class
```
src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTestCase.php
```
which is in the test folder so this is not installed in the user app.

On the other side, there is 
```
src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php
```
which is in the user app, and for this one I have already checked 
```
if (class_exists(LockMiddleware::class) && class_exists(LockFactory::class)) {
```
cf
https://github.com/symfony/symfony/pull/54141/files#diff-e4f52d5033296421d02e360c8355aa1931bdfb0c12579f4815ea02b724322c0cR2179

Reply:
Aaah sorry I missed the "Test" part in the classname...